test/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter_test.cc

```diff
@@ -1224,6 +1224,25 @@ TEST_F(GrpcJsonTranscoderFilterConvertGrpcStatusTest,
   EXPECT_FALSE(response_headers.has("grpc-status-details-bin"));
 }
 
+TEST_F(GrpcJsonTranscoderFilterConvertGrpcStatusTest,
+       TranscodingPercentEncodedTextHeadersInTrailerOnlyResponse) {
+  const std::string expected_response(R"({"code":5,"message":"Resource not found"})");
+  EXPECT_CALL(encoder_callbacks_, addEncodedData(_, false))
+      .WillOnce(Invoke([&expected_response](Buffer::Instance& data, bool) {
+        EXPECT_EQ(expected_response, data.toString());
+      }));
+
+  Http::TestResponseHeaderMapImpl response_headers{{":status", "200"},
+                                                   {"content-type", "application/grpc"},
+                                                   {"grpc-status", "5"},
+                                                   {"grpc-message", "Resource%20not%20found"}};
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_.encodeHeaders(response_headers, true));
```

Comment:
Do you think these checks are necessary here? We are only verifying that percent-encoded grpc status has been decoded; the other checks have been performed elsewhere and imo only muddle this test.

Reply:
I think that these redundant tests aren't necessarily bad.